tools/hidl/hidlcomp.h

```diff
@@ -989,6 +996,29 @@ class EspMethodInfo
         response_ =strdup(name);
     }
 
+    bool isExecutionProfilingEnabled() const {return enableExecutionProfiling;}
+    void setEnableExecutionProfiling(bool enable)
+    {
+        enableExecutionProfiling = enable;
+    }
+
+    const std::string &getExecutionProfilingOptions(){return executionProfilingOptions;}
+    void setExecutionProfilingOptions(const char *options)
+    {
+        executionProfilingOptions = std::string(options);
+        auto noSpaceEnd = std::remove(executionProfilingOptions.begin(), executionProfilingOptions.end(), ' ');
+        executionProfilingOptions.erase(noSpaceEnd, executionProfilingOptions.end());
+        validateProfileExecutionOptions(executionProfilingOptions);
+    }
+
+    std::string getExecutionProfilingMetricName()
+    {
+        std::string name("p");
```

Comment:
You could save the qualified name in the EspMethodInfo class rather than building it every time.